### PR TITLE
WIP: Initializing the advance-channel protocol

### PR DESCRIPTION
**Pull request by andrewgordstewart**
I think that the initialization logic is complete, but it is not yet tested for the hub. I am adding tests for the scenarios in order.

### CODE REVIEW

File: packages/wallet/src/redux/protocols/advance-channel/__tests__/reducer.test.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+) => {
+  it(`transitions to ${type}`, () => {
+    expect(result.type).toEqual(type);
+  });
+};
+
+describe('sending preFundSetup as A', () => {
+  const scenario = scenarios.newChannelAsA;
+
+  describe.only('when initializing', () => {
```
Comment (andrewgordstewart) [inline packages/wallet/src/redux/protocols/advance-channel/__tests__/reducer.test.ts]: The tests are not fully written, but I think the scenarios are mostly complete.

Only this test is enabled.

File: packages/wallet/src/redux/protocols/advance-channel/__tests__/scenarios.ts [hunk 2]
```diff
@@ -53,6 +47,18 @@ const propsB = {
   ourIndex: ThreePartyPlayerIndex.B,
 };
 
+const propsHub = {
+  ...props,
+  ourIndex: ThreePartyPlayerIndex.Hub,
+};
+
+const commitments0 = [signedCommitment0];
```
Comment (kerzhner) [inline packages/wallet/src/redux/protocols/advance-channel/__tests__/scenarios.ts]: Do you think there is a more descriptive yet succinct variable name that describes the channel state for the sequence of commitments?

File: packages/wallet/src/redux/__tests__/helpers.ts [hunk 3]
```diff
@@ -79,9 +79,35 @@ export const expectThisMessageAndCommitmentSent = (
...
+
+export const expectThisMessageAndTheseCommitmentsSent = (
+  state: SideEffectState,
+  commitments: PartialCommitments,
+  messageType: string,
+) => {
+  expectSideEffect('messageOutbox', state, item => {
+    expect(item.messagePayload.type).toEqual(messageType);
+    expect(item.messagePayload.signedCommitments).toMatchObject(
+      commitments.map(({ commitment }) => ({ commitment })), // This has the effect of ignoring the signature
```
Comment (kerzhner) [inline packages/wallet/src/redux/__tests__/helpers.ts]: Can't wrap my head around this...

Comment (andrewgordstewart) [reply packages/wallet/src/redux/__tests__/helpers.ts]: `commitments` are signed (partial) commitments, so they have the shape `{ commitment: Commitment, signature: string }`

This helper is only concerned with the commitment part of it, and doesn't assert anything about the signature (in keeping with the existing helper for a single commitment)

File: packages/wallet/src/redux/protocols/advance-channel/__tests__/scenarios.ts (hunk 2 above)
Comment (andrewgordstewart) [reply packages/wallet/src/redux/protocols/advance-channel/__tests__/scenarios.ts]: Perhaps, but I'm far ahead of origin's HEAD locally, so I would prefer not to rename it in this PR.

**Pull request merged**